### ISSUE TITLE
[HOLD for payment 2023-11-29] [DISTANCE] [$500] LOW: The notification in Admin is inconsistent with the error message in Reimbursements Section

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#27720]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to settings>workspace> Reimbursement.
2. Enter 0 in the Reimbursement field and observe that it is not allowed.
3. Add 0.001 to the Reimbursement field.
4. Check the admin section and notice that the notification shows 0.00.
5. Note that the system rounds the value of 0.001 to 0.00.
6. Since 0 is not allowed in the Reimbursement field, the notification should not show 0.00.
## Expected Result: 
The notification should be consistent with the error message.
## Actual Result: 
The notification is inconsistent with the error message.
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.71-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/996d81eb-9c49-4109-ac7f-7b37702a159c

https://github.com/Expensify/App/assets/43996225/76149eec-8e34-4740-8fda-3eef26b86149

**Expensify/Expensify Issue URL:**
**Issue reported by:** @tewodrosGirmaA
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1695012404635409

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012843502c5f0d6cbe</li>
        <li>Upwork Job ID: 1703890469929783296</li>
        <li>Last Price Increase: 2023-10-09</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @peterdbarkerUK</details>

**Comment by paultsimura** [comment]
BE issue, this comes in response to the `UpdateWorkspaceCustomUnitAndRate` operation:

<img width="756" alt="image" src="https://github.com/Expensify/App/assets/12595293/dc93a0d2-a40c-4f60-9548-088c7412628b">

**Comment by nebiyuelias1** [comment]
> BE issue, this comes in response to the `UpdateWorkspaceCustomUnitAndRate` operation:
> 
> <img alt="image" width="756" src="https://user-images.githubusercontent.com/12595293/268791548-dc93a0d2-a40c-4f60-9548-088c7412628b.png">

But what about the payload?

**Comment by peterdbarkerUK** [comment]
Reproduced
<img width="604" alt="image" src="https://github.com/Expensify/App/assets/16888060/2716677c-2962-45b2-aa5f-54f06d62b513">

**Comment by peterdbarkerUK** [comment]
kind of a dupe of [this](https://github.com/Expensify/App/issues/26344), opened a discussion [in slack](https://expensify.slack.com/archives/C01SKUP7QR0/p1695108866203249) to confirm next steps

**Comment by peterdbarkerUK** [comment]
Yeah that's fair. Do you think this is best fixed via a separate PR, or by updating [this one](https://github.com/Expensify/App/pull/26745)?

**Comment by peterdbarkerUK** [comment]
I'm sorry, I was thinking from a payment perspective, not from an
engineering perspective - and not being helpful.

Ignore me, let's just deal with this separately :)

Op do 21 sep. 2023 11:58 schreef Tienifr ***@***.***>:

**Comment by peterdbarkerUK** [comment]
@abdulrahuman5196 - what do you think of [this proposal](https://github.com/Expensify/App/issues/27720#issuecomment-1724845761)?

**Comment by peterdbarkerUK** [comment]
@abdulrahuman5196 - could you take a look at @tienifr proposal?

**Comment by abdulrahuman5196** [comment]
Reviewing in couple of hours

**Comment by abdulrahuman5196** [comment]
Rechecking again

**Comment by abdulrahuman5196** [comment]
@peterdbarkerUK I think, if we allow 3 decimal points in the input we will do the same in the notification text. Or we should update the input so that it doesn't allow invalid decimal points.

But I am not sure on the expectation. It would be better to check with an internal engineer worked on distance to understand what we should do here. Can we check internally on this first?

**Comment by peterdbarkerUK** [comment]
@abdulrahuman5196 - I agree with @tienifr that the problem is 
>The notification is inconsistent with the error message.

The mileage system correctly accounts for decimals depending on currency, only the notification here does not, right?

I don't think it's the intention to prevent three-decimal (or other non-two-decimal) place inputs, as this is valid for certain currencies.

**Comment by abdulrahuman5196** [comment]
@peterdbarkerUK Then could we loop in an internal engineer on this? If the notification is to be changed?

**Comment by abdulrahuman5196** [comment]
@tienifr I agree on the above comment. But this need more checking than that, because this value directly affects distance requests as well.
But in distance requests we are allowing 3 decimal points. And in terms of notification, for other currencies i am not seeing proper notifications.

IMO, We should first confirm internally on what place and behaviour we should change? Should it be same as split bill or something else? And then work on the fix behaviour.

![Screenshot (Oct 4, 2023 2 59 19 PM)](https://github.com/Expensify/App/assets/46707890/a146ed0f-8bd8-48b5-b00a-965d4f35641f)

**Comment by abdulrahuman5196** [comment]
@peterdbarkerUK This is waiting on your feedback?

**Comment by peterdbarkerUK** [comment]
Hmm, I think we are all saying the same thing here:

>@peterdbarkerUK it's valid for some currencies but we should only allow that for those currencies.

>I think, if we allow 3 decimal points in the input we will do the same in the notification text.

I.e. we all agree with @tienifr's proposal, right?

**Comment by abdulrahuman5196** [comment]
@peterdbarkerUK Yes I like the proposal. But I need confirmation from internal if we expect same behaviour as in proposal?
Since its used in distance request as well. Could you kindly check on this comment? https://github.com/Expensify/App/issues/27720#issuecomment-1746497073

**Comment by peterdbarkerUK** [comment]
Sure, I've raised it in  [internal slack](https://expensify.slack.com/archives/C01SKUP7QR0/p1697143653544339) :)

**Comment by peterdbarkerUK** [comment]
@neil-marcellini gonna have a look-see :)

**Comment by neil-marcellini** [comment]
@abdulrahuman5196 I like @tienifr's [proposal](https://github.com/Expensify/App/issues/27720#issuecomment-1724845761). We discussed internally and agree that the number of decimals shown should be based on the rate. The backend doesn't really do any validation on the rate as far as I can tell, but it only makes sense to match the currency.

**Comment by neil-marcellini** [comment]
Lmk if we need any backend changes, but I think it might work without any.

**Comment by abdulrahuman5196** [comment]
Thank you for the confirmation @neil-marcellini  https://github.com/Expensify/App/issues/27720#issuecomment-1769471952

Anyways approving again from my end as well. @tienifr 's proposal here https://github.com/Expensify/App/issues/27720#issuecomment-1724845761 looks good to me as well. 

🎀 👀 🎀
C+ Reviewed

**Comment by neil-marcellini** [comment]
Heads up @tienifr we discussed a bit more and came away with a couple extra ideas. The IRS mileage rate in the US is expressed in fractions of a cent for increased accuracy, so we want to keep that behavior.

- allow one more decimal place than the number of decimal places in the currency
- format the rate in the notification in the same way

**Comment by abdulrahuman5196** [comment]
@tienifr Could you check and update on the above?

**Comment by abdulrahuman5196** [comment]
@tienifr Agree on the above. For this PR we can target frontend fix and the @neil-marcellini  can check on the backend notification fix.

**Comment by neil-marcellini** [comment]
Rodrigo is taking over now, but please DM me on NewDot if you have any questions about our previous discussions.

**Comment by rlinoz** [comment]
Just to make sure, @tienifr is the one working on the frontend fix?

**Comment by rlinoz** [comment]
@tienifr BE fix is deployed to staging.

**Comment by bondydaa** [comment]
just FYI we had to revert the changes in the backend because they were causing errors internally.

**Comment by rlinoz** [comment]
BE fix just got merged, as soon as it is deployed to staging I will notify @tienifr

**Comment by rlinoz** [comment]
Hey @tienifr the fix is in staging now, if you want to start the FE fix.

**Comment by rlinoz** [comment]
It is in production as well now.

**Comment by abdulrahuman5196** [comment]
The BE changes are deployed, @tienifr Will be working on PR

**Comment by abdulrahuman5196** [comment]
The BE changes are deployed, @tienifr Will be working on PR

**Comment by rlinoz** [comment]
If there is someone with the star next to its name it has to be that person to comment to remove the overdue label from now on.

Anyway, as @abdulrahuman5196 said @tienifr will be working on this PR.

**Comment by abdulrahuman5196** [comment]
>The PR that introduced the bug has been identified. Link to the PR:
The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not a regression. Seems to be the case from before.

>Determine if we should create a regression test for this bug.

Yes

>If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

Expected behavior: Allow one more decimal place for currency decimal. For example, JPY's decimals value is 0, we should allow up-to-1-decimal values like JPY 1, JPY 0.1, but not JPY 0.01.

1) Go to Settings >> Workspace >> Settings >> Default currency
2) Change to USD - $
3) Go to Settings >> Workspace >> Reimbursements >> Track distance Rate
4) Enter a 4-decimal value (i.e. 0.0001) >> Save
5) Verify an error appears
6) Enter a 3-decimal value (i.e. 0.001) >> Save
7) Verify that in #admin room, a message shows the exact amount and currency from step 6 (i.e. $0.001)
8) Repeat step 1 to 7 for other currencies and their corresponding decimals


@peterdbarkerUK

**Comment by peterdbarkerUK** [comment]
Payment Summary:
- @abdulrahuman5196 $500 C+ ([Offer](https://www.upwork.com/nx/wm/offer/27886545))
- @tienifr $500 Contributor ([Offer](https://www.upwork.com/nx/wm/offer/27886576))
- @tewodrosGirmaA $50 Reporter ([Offer](https://www.upwork.com/nx/wm/offer/27886607))

**Comment by rlinoz** [comment]
Do we have anything else to do here @peterdbarkerUK ?

**Comment by peterdbarkerUK** [comment]
Apologies, was OoO most of Friday. All paid and contracts closed!
